src/record-error-archival/src/recordErrorArchival.ts

```diff
@@ -0,0 +1,63 @@
+import type { ApiClient } from "shared-types"
+import { AuditLogEvent, isError, isSuccess } from "shared-types"
+import type { ArchivedErrorRecord } from "./DatabaseClient"
+import type DatabaseClient from "./DatabaseClient"
+import groupBy from "lodash.groupby"
+import { logger } from "shared"
+
+export const recordErrorArchival = async (db: DatabaseClient, api: ApiClient) => {
+  await db.connect()
+  const errorRecords = await db.fetchUnloggedArchivedErrors()
+
+  const results: { success: boolean; reason?: string; errorRecord: ArchivedErrorRecord }[] = []
+
+  for (const errorRecord of errorRecords) {
+    logger.debug({ message: "Would audit log the archival of an error", record: errorRecord })
+
+    const auditLogEvent = new AuditLogEvent({
+      eventSource: errorRecord.archivedBy,
+      category: "information",
+      eventType: "Error archival",
+      timestamp: errorRecord.archivedAt
+    })
+    auditLogEvent.addAttribute("Error ID", errorRecord.errorId)
+
+    const response = await api.createEvent(errorRecord.messageId, auditLogEvent)
+    if (isError(response)) {
+      logger.error({
+        message: "Failed to add archived error to audit log",
+        reason: response.message,
+        record: errorRecord
+      })
+    }
+
+    results.push({
+      success: isSuccess(response),
+      reason: isError(response) ? response.message : undefined,
+      errorRecord: errorRecord
+    })
+  }
+
+  // Mark successfully audit logged errors
+  const successful = results.filter((result) => result.success).map((result) => result.errorRecord.errorId)
+  if (successful.length > 0) {
+    await db.markErrorsAuditLogged(successful)
```

Comment:
We should also check if this function was not successful.
Also, assuming that it keeps failing to update the records in postgres which results in lambda failure or timeout, we should make sure that in the next lambda invocation, we don't insert the events to audit log records again. (to avoid duplication)